test/Elastic.Apm.DockerTests/ContainerIdCalculationTests.cs

```diff
@@ -0,0 +1,44 @@
+using System.IO;
+using System.Text;
+using Elastic.Apm.Helpers;
+using Elastic.Apm.Tests.Mocks;
+using FluentAssertions;
+using Xunit;
+
+namespace Elastic.Apm.DockerTests
+{
+	public class ContainerIdCalculationTests
+	{
+		[Theory]
+		[InlineData("13:name=systemd:/docker/ae8d0ef9c8757a42d01e994b0ffe0e93e26ece256f68cd850467836c83202367",
+			"ae8d0ef9c8757a42d01e994b0ffe0e93e26ece256f68cd850467836c83202367")]
+		[InlineData("12:devices:/docker/051e2ee0bce99116029a13df4a9e943137f19f957f38ac02d6bad96f9b700f76",
+			"051e2ee0bce99116029a13df4a9e943137f19f957f38ac02d6bad96f9b700f76")]
+		[InlineData("1:name=systemd:/system.slice/docker-cde7c2bab394630a42d73dc610b9c57415dced996106665d427f6d0566594411.scope",
+			"cde7c2bab394630a42d73dc610b9c57415dced996106665d427f6d0566594411")]
+		[InlineData("1:name=systemd:/kubepods/besteffort/pode9b90526-f47d-11e8-b2a5-080027b9f4fb/15aa6e53-b09a-40c7-8558-c6c31e36c88a",
+			"15aa6e53-b09a-40c7-8558-c6c31e36c88a")]
```

Comment:
I would also add the following test mentioned at https://github.com/elastic/apm/issues/21#issuecomment-444410552

    "1:name=systemd:/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod90d81341_92de_11e7_8cf2_507b9d4141fa.slice/crio-2227daf62df6694645fee5df53c1f91271546a9560e8600a525690ae252b7f63.scope": {
        "container": {
            "id": "2227daf62df6694645fee5df53c1f91271546a9560e8600a525690ae252b7f63",
        },
        "pod": {
            "uid": "90d81341_92de_11e7_8cf2_507b9d4141fa",
        }
    }

Reply:
👍 added